benches/sparse.rs

```diff
@@ -0,0 +1,190 @@
+#![feature(test)]
+
+extern crate test;
+
+use bio::alignment::sparse::*;
+use test::Bencher;
+
+static K_FACTOR: usize = 50;
+
+static MATCHES: &'static [(u32, u32)] = &[(0, 1000), (1000, 2000), (2000, 3000), (3000, 4000),
```

Comment:
This set of matches is quite small, and there's 'branching' of the possible paths, which means some of the code paths in the DP won't be exercised. Might be interesting to generate a set of matches from a longer or repetitive sequence to exercise the DP a bit more.

Reply:
I would be happy to expand the list of matches and get more applicable  sequence data. How would you recommend creating a longer set of matches? These tuples represent intervals over the sequence to analyze, correct? 

Also, what tools do you use to generate these test sequences? I notice they have a specific number of base pairs and nucleotide region composition.

Reply:
A very basic sequence simulator is included with the read simulator Mason from Seqan:
https://www.seqan.de/apps/mason/

Easiest install is probably [via bioconda](https://bioconda.github.io/user/install.html#install-conda):
https://bioconda.github.io/recipes/mason/README.html

The command to use is `mason_genome`, simply check how to use it with `mason_genome --help`.
As for repetitions in the sequence, you would have to manually create them after simulating the genome.

But maybe someone else knows a more versatile simulator for DNA sequences, that allows for controlling things like repetitions right away?

Reply:
I updated the code to include a longer set of tuples in the 'matches' array. I also generated two 6000-base-pair random sequences, formatted according to other benchmark source code files. Is this a sufficient size for the 'matches' array?